### PR TITLE
Fix #9. Change logging level.

### LINKED ISSUE #9: Race condition leads to dark mode not being applied
## Bug description

@lionirdeadman reported a bug with a specific website that triggered 

## Steps to reproduce

Dark mode only used in pure CSS. Same CSS on each site.

But on sites with much HTML content,

I currently guess this is a timing issue, because the add-on tries to get the CSS from the website too fast, so it has not already loaded (because it's still parsing HTML).

### Actual behavior

Sometimes fails to apply dark mode.

### Expected behavior

Always work.

## System

Browser and version: Firefox 68

## Possible solution

trigger later?

### PR DESCRIPTION
**Pull request by kmcopper**
Change the default logging level to allow better filtering. Fix the #9 race condition by waiting fully for the page to load rather then just for the DOM to load.

**Edit (by @rugk):** Fixes #9

### CODE REVIEW

File: src/content_scripts/applyColorSchemeCss.js [hunk 1]
```diff
@@ -170,5 +170,5 @@ async function applyWantedStyle() { // eslint-disable-line no-unused-vars
     });
 }
 
-// apply style when DOM content is loaded
-document.addEventListener("DOMContentLoaded", applyWantedStyle);
+// apply style when content is fully loaded
+window.addEventListener('load', applyWantedStyle);
```
Comment (rugk) [inline src/content_scripts/applyColorSchemeCss.js]: please see eslint/code styles:

```suggestion
window.addEventListener("load", applyWantedStyle);
```

**Comment by rugk** [comment]
Okay tested it with the test website to reproduce in https://github.com/rugk/website-dark-mode-switcher/issues/9 and it did indeed work/fix the issue. So thanks for tackling this issue, especially even if you did not have the site/exact STR.

---

However, there is one issue I could imagine: performance.

As this triggers later now, for all other websites the delay that the website is still displayed in "light mode" may be a bit too much/long.
This is just an assumption/theory and needs testing though, I did not compare it yet.
As such, what do you think of the idea to combine these two triggers, and just run them both/only run the second if the first "make website black" fails?

(Just an idea, as said.)

**Comment by kmcopper** [comment]
The first method is indeed significantly quicker in some cases so I implemented dual triggers as suggested. 

It's sometimes noticeably quicker but difficult to reproduce. I managed to find a site of my own with this issue and it triggered me enough into troubleshooting and finding a fix for it at the source rather then attempting to blame the site operator for it since it mostly seemed other sites worked fine. This whole prefers-color-scheme business is confusing so I was quite unsure rather to blame firefox, the site, or the extension until I found that issue.

File: src/content_scripts/applyColorSchemeCss.js [hunk 2]
```diff
@@ -170,5 +173,13 @@ async function applyWantedStyle() { // eslint-disable-line no-unused-vars
     });
 }
 
+async function applyStyleFailed() {
```
Comment (rugk) [inline src/content_scripts/applyColorSchemeCss.js]: See eslint, you shall add a JSDOC for such a function. :smiley:

File: src/content_scripts/applyColorSchemeCss.js [hunk 3]
```diff
@@ -12,6 +12,8 @@ const COMMUNICATE_REMOVE_CSS = "removeCss";
 
 // need to save injected CSS, so we can remove it later
 let injectedCss = "";
+// placeholder variable for if the style has failed
+let styledFailed = false;
```
Comment (rugk) [inline src/content_scripts/applyColorSchemeCss.js]: Maybe this is a better name?
```suggestion
let applyStyleFailed = false;
```

**Comment by kmcopper** [comment]
As far as I'm aware the load event actually only fires after the DOMContentLoaded event in exact order but I'm neither a web developer nor an extension developer so I'm not exactly sure. The worst that could happen while I was playing with it was a flicker and double paint. But I've not seen it happen with this revision. We can always revisit if an issue crops up at a later date.

**Comment by rugk** [comment]
Good and thanks again for the PR. :smiley:

**Pull request merged**